catalog/message/schema/catalog-schema.json

```diff
@@ -1,39 +1,381 @@
+                    "$ref": "#/definitions/time"
+                },
+                "dct:publisher": {
+                    "$ref": "#/definitions/Reference"
+                },
+                "dct:relation": {
+                    "$ref": "#/definitions/Reference"
+                },
+                "dct:title": {
+                    "type": "string"
+                },
+                "dct:type": {
+                    "$ref": "#/definitions/Reference"
+                },
+                "odrl:hasPolicy": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "https://raw.githubusercontent.com/International-Data-Spaces-Association/ids-specification/main/negotiation/message/schema/contract-schema.json#/definitions/Policy"
+                    },
+                    "minItems": 1
+                }
+            }
+        },
+        "Dataset": {
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/Resource"
+                },
+                {
+                    "properties": {
+                        "dcat:distribution": {
+                            "type": "array",
+                            "items": {
+                                "$ref": "#/definitions/Distribution"
+                            },
+                            "minItems": 1
+                        },
+                        "dcat:spatialResolutionInMeters": {
+                            "$ref": "#/definitions/Reference"
+                        },
+                        "dcat:temporalResolution": {
+                            "$ref": "#/definitions/duration"
+                        },
+                        "dct:accrualPeriodicity": {
+                            "$ref": "#/definitions/Reference"
+                        },
+                        "dct:spatial": {
+                            "$ref": "#/definitions/Reference"
+                        },
+                        "dct:temporal": {
+                            "$ref": "#/definitions/Reference"
+                        },
+                        "prov:wasGeneratedBy": {
+                            "$ref": "#/definitions/Reference"
                         }
+                    }
+                }
+            ]
+        },
+        "DataService": {
+            "type": "object",
+            "properties": {
+                "dcat:endpointDescription": {
+                    "$ref": "#/definitions/Reference"
+                },
+                "dcat:endpointURL": {
+                    "$ref": "#/definitions/Reference"
+                },
+                "dcat:servesDataset": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Dataset"
+                    },
+                    "minItems": 1
+                }
+            }
+        },
+        "Distribution": {
```

Comment:
Shouldn't all classes also have a `@type` property (same goes for DataService, Dataset, Resource)

Reply:
Not necessarily. As long as it's not at the beginning of a JSON, the serialiser usually knows the class through the property (JSON key). However, and that's then a real problem, in case several different classes (subclasses) are allowed, this approach won't work. On the other hand, I'd say that allowing different classes is a bad practice for our case anyway...

Reply:
I agree, but don't we already have this problem with `dct:hasPart` in Catalog which is either a Catalog, Dataset, or DataService?
So in that case we would need it for some objects and then it might be easier to include the `@type` everywhere to not require complex logic to see where to include it and where not.

Reply:
Moved to #160 to track it independently.